raiden/transfer/architecture.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=too-few-public-methods
+import types
+from collections import namedtuple
+from copy import deepcopy
+
+Iteration = namedtuple('Iteration', ('new_state', 'events'))
```

Comment:
I would prefer `StateTransition` since it sounds more logical to me , but  `TransitionResult` is also nice.